tests/src/test/scala/akka/kafka/internal/ConsumerProgressTrackingSpec.scala

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright (C) 2014 - 2016 Softwaremill <https://softwaremill.com>
+ * Copyright (C) 2016 - 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.kafka.internal
+
+import akka.kafka.tests.scaladsl.LogCapturing
+import org.apache.kafka.clients.consumer.{Consumer, ConsumerRecord, ConsumerRecords, OffsetAndMetadata}
+import org.apache.kafka.common.TopicPartition
+import org.mockito.Mockito
+import org.scalatest.flatspec.AnyFlatSpecLike
+import org.scalatest.matchers.should.Matchers
+
+import scala.jdk.CollectionConverters._
+import scala.language.reflectiveCalls
+
+class ConsumerProgressTrackingSpec extends AnyFlatSpecLike with Matchers with LogCapturing {
+
+  private val tp = new TopicPartition("t", 0)
+  private val m1 = new ConsumerRecord[String, String](tp.topic(), tp.partition(), 10L, "k1", "kv")
+  def asConsumerRecords[K, V](tp: TopicPartition, records: ConsumerRecord[K, V]*): ConsumerRecords[K, V] = {
+    new ConsumerRecords[K, V](Map(tp -> records.asJava).asJava)
+  }
+  private val records = asConsumerRecords(tp, m1)
+
+  private def extractOffsetFromSafe(entry: (TopicPartition, SafeOffsetAndTimestamp)) = (entry._1, entry._2.offset)
+  private def extractOffset(entry: (TopicPartition, OffsetAndMetadata)) = (entry._1, entry._2.offset())
+
+  it should "add requested offsets" in {
+    val tracker = new ConsumerProgressTrackerImpl()
+    tracker.commitRequested shouldBe empty
+    tracker.committedOffsets shouldBe empty
+    tracker.commitRequested(Map(tp -> offset(0)))
+    tracker.commitRequested should have size 1
+    tracker.commitRequested(tp).offset() should be(0)
+    tracker.receivedMessages shouldBe empty
+    tracker.committedOffsets shouldBe empty
+  }
+
+  it should "track received records" in {
+    val tracker = new ConsumerProgressTrackerImpl()
+    tracker.assignedPositions(Set(tp), Map(tp -> 0L))
+    tracker.received(records)
+
+    tracker.commitRequested(tp).offset() should be(0)
+    tracker.receivedMessages should have size 1
+    tracker.receivedMessages(tp).offset should be(10L)
+    tracker.receivedMessages(tp).timestamp should be(ConsumerRecord.NO_TIMESTAMP)
+  }
+
+  it should "filter out non-assigned partitions in received messages" in {
+    val tracker = new ConsumerProgressTrackerImpl()
+    // received, but wasn't requested (assigned), so empty received/requested/committed
+    tracker.received(asConsumerRecords(tp, m1))
+    tracker.commitRequested shouldBe empty
+    tracker.receivedMessages shouldBe empty
+    tracker.committedOffsets shouldBe empty
+
+    // once is is assigned, we should start tracking it
+    tracker.assignedPositions(Set(tp), Map(tp -> 0L))
+    // haven't received any records yet, stays empty
+    tracker.receivedMessages shouldBe empty
+    tracker.commitRequested.map(extractOffset) should be(Map(tp -> 0L))
+    tracker.commitRequested.map(extractOffset) should be(Map(tp -> 0L))
```

Comment:
I think meant to be:

```suggestion
    tracker.committedOffsets.map(extractOffset) should be(Map(tp -> 0L))
```